control_toolbox/include/control_toolbox/low_pass_filter.hpp

```diff
@@ -143,12 +151,18 @@ LowPassFilter<T>::~LowPassFilter()
 template <typename T>
 bool LowPassFilter<T>::configure()
 {
-  // Initialize storage Vectors
-  filtered_value = filtered_old_value = old_value = std::numeric_limits<T>::quiet_NaN();
-  // TODO(destogl): make the size parameterizable and more intelligent is using complex types
-  for (Eigen::Index i = 0; i < 6; ++i)
+  // Initialize storage Vectors, depending on the type of T
+  if constexpr (std::is_same<T, geometry_msgs::msg::WrenchStamped>::value)
+  {
+    // TODO(destogl): make the size parameterizable and more intelligent is using complex types
+    for (Eigen::Index i = 0; i < 6; ++i)
+    {
+      filtered_value_[i] = filtered_old_value_[i] = old_value_[i] = std::numeric_limits<double>::quiet_NaN();
+    }
```

Comment:
This can be completely skipped, if we define EIGEN_INITIALIZE_MATRICES_BY_NAN before including Eigen. @christophfroehlich what do you think about it?. I don't have a strong opinion here

Reply:
didn't know about that, why not